code/game/objects/items/weapons/pneumaticCannon.dm

```diff
@@ -85,20 +101,29 @@
 	if(tank && !tank.air_contents.remove(gasPerThrow * pressureSetting))
 		user << "<span class='warning'>\The [src] lets out a weak hiss and doesn't react!</span>"
 		return
-	user.visible_message("<span class='danger'>[user] fires \the [src]!</span>", \
-			     "<span class='warning'>You fire \the [src]!</span>")
-	add_logs(user, target, "fired at", src)
-	playsound(src.loc, 'sound/weapons/sonic_jackhammer.ogg', (50 * pressureSetting), 1)
+	if(user.disabilities & CLUMSY && prob(25))
+		user.visible_message("<span class='warning'>[user] loses their grip on [src], causing it to go off!</span>", \
+							 "<span class='userdanger'>[src] slips out of your hands and goes off!</span>")
+		user.drop_item()
+		if(prob(10))
+			target_turf = get_turf(user)
+		else
+			for(var/turf/T in range(3,src))
+				if(prob(25))
+					target_turf = T
```

Comment:
should put a break after this, so that when it finds a turf it stops the loop
alternatively, ```var/list/possible_targets = range(3,src) target_turf = pick(T)```